packages/api/src/providers/fitbit.ts

```diff
@@ -444,48 +444,100 @@ export class Fitbit extends Provider implements OAuth2 {
             headers: {
               Authorization: `Bearer ${accessToken}`,
             },
           });
-        })
-      );
-    } catch (err) {
-      console.log(`Failed to delete existing Fitbit WH subscriptions. Cause: ${err}`);
-      capture.error("Failed to delete existing Fitbit WH subscriptions", {
+        } catch (err) {
+          rejected.push({ url, err });
+        }
+      })
+    );
+    if (rejected.length > 0) {
+      console.log(`Failed to delete existing Fitbit WH subscriptions.`);
+      capture.message("Failed to delete existing Fitbit WH subscriptions", {
         extra: { context: "fitbit.deleteSubscriptions" },
       });
-      throw new Error("Failed to delete active WH subscriptions for Fitbit", { cause: err });
+      throw new Error("Failed to delete active WH subscriptions for Fitbit", { cause: rejected });
     }
   }
 
   /**
    * Fetches existing subscriptions for the Fitbit user, and returns an array of subscribed collectionTypes
    *
-   * @param accessToken Authorization Bearer-type token
-   * @returns List of active FitbitWebhookSubscriptions or undefined
+   * @param token: FitbitToken
+   * @returns List of active FitbitWebhookSubscriptions
    */
-  async getActiveSubscriptions(
-    accessToken: string
-  ): Promise<FitbitWebhookSubscriptions | undefined> {
-    const url = `${Fitbit.URL}/${Fitbit.API_PATH}/apiSubscriptions.json`;
-    try {
-      const resp = await axios.get(url, {
-        headers: {
-          Authorization: `Bearer ${accessToken}`,
-        },
+  async getActiveSubscriptions(token: FitbitToken): Promise<FitbitWebhookSubscriptions> {
+    const rejected: { url: string; err: unknown }[] = [];
+    const activeSubscriptions: FitbitWebhookSubscriptions[] = [];
+    await Promise.allSettled(
+      Object.entries(Fitbit.subscriptionTypes).map(async ([scope, collectionType]) => {
+        if (token.scope.includes(scope)) {
+          const url = `${Fitbit.URL}/${Fitbit.API_PATH}/${collectionType}/apiSubscriptions.json`;
+          try {
+            const resp = await axios.get(url, {
+              headers: {
+                Authorization: `Bearer ${token.accessToken}`,
+              },
+            });
+            if (resp.status === status.OK && resp.data.apiSubscriptions.length) {
+              const activeSubscription = fitbitWebhookSubscriptionsSchema.parse(
+                resp.data.apiSubscriptions
+              );
+              activeSubscriptions.push(activeSubscription);
+            }
+          } catch (err) {
+            rejected.push({ url, err });
+          }
+        }
+      })
+    );
+    if (rejected.length > 0) {
+      console.log(`Failed to get active Fitbit WH subscriptions.`);
+      capture.message(`Failed to get active Fitbit WH subscriptions.`, {
+        extra: { context: `fitbit.getActiveSubscriptions`, cause: rejected },
+      });
+      throw new Error("Failed to get active Fitbit WH subscriptions", { cause: rejected });
+    }
```

Comment:
Same thing here WRT handling errors downstream and upstream. We're reporting to Sentry twice.

Also, as I mentioned, `cause` is supposed to be an `Error`, not some arbitrary object like we have here (`{ url: string; err: any }[]`). If we really need additional info on an Error, we can use `MetriportError`.

Finally, wondering if we can just log/report the error and not throw...